src/query.rs

```diff
@@ -214,3 +230,50 @@ pub fn q_once<'sqlite, 'schema, 'query, T>
 
     run_algebrized_query(sqlite, algebrized)
 }
+
+pub fn q_explain<'sqlite, 'schema, 'query, T>
+(sqlite: &'sqlite rusqlite::Connection,
+ schema: &'schema Schema,
+ query: &'query str,
+ inputs: T) -> Result<QueryExplanation>
+        where T: Into<Option<QueryInputs>>
+{
+    let parsed = parse_find_string(query)?;
+    let algebrized = algebrize_with_inputs(schema, parsed, 0, inputs.into().unwrap_or(QueryInputs::default()))?;
+
+    let unbound = algebrized.unbound_variables();
+    if !unbound.is_empty() {
+        bail!(ErrorKind::UnboundVariables(unbound.into_iter().map(|v| v.to_string()).collect()));
+    }
+
+    let select = query_to_select(algebrized)?;
+    let query = select.query.to_sql_query()?;
+
+    let plan_sql = format!("EXPLAIN QUERY PLAN {}", query.sql);
+
+    let mut statement = sqlite.prepare(plan_sql.as_str())?;
+
+    let from_row = |row: &rusqlite::Row| {
+        QueryPlanStep {
+            select_id: row.get(0),
+            order: row.get(1),
+            from: row.get(2),
+            detail: row.get(3)
+        }
+    };
+
+    let mut rows = if query.args.is_empty() {
+        statement.query_map(&[], from_row)?
+    } else {
+        let refs: Vec<(&str, &ToSql)> =
+            query.args.iter()
+                .map(|&(ref k, ref v)| (k.as_str(), v.as_ref() as &ToSql))
```

Comment:
Stack your dots. Or, better, factor this logic out of `run_algebrized_query`! Everything up to `query_to_select` is shared, and so is this logic for running some SQL.

You'll need to use `query` and `query_named` instead of the `_map_` variant, but that's OK, I think.

Reply:
I think putting it into `run_algebrized_query` will result in a lot of conditionals in that function to support both cases. I could probably pull the common parts into something separate though.